flight_backend/api/views/users.py

```diff
@@ -45,3 +45,29 @@ def post(self, request):
             dict(message="email, password, firstname and lastname are required"),
             status=status.HTTP_400_BAD_REQUEST,
         )
+
+
+class UserLoginViewSet(generics.CreateAPIView):
+    permission_classes = (AllowAny,)
+
+    def post(self, request):
+        authenticated_user = validate_login_details(request.data)
+        if authenticated_user:
+            serializer = UserLoginSerializer(authenticated_user)
+            token_serializer = TokenSerializer(data={
+                "token": jwt_encode_handler(
+                    jwt_payload_handler(get_user(serializer.data.get('id')))
+                )
+            })
+            if token_serializer.is_valid():
+                response = {
+                    'token': token_serializer.data.get('token'),
+                    'id': serializer.data.get('id'),
+                    'first_name': serializer.data.get('first_name'),
+                    'last_name': serializer.data.get('last_name'),
+                    'email': serializer.data.get('email')
+                }
+                return Response(response, status=status.HTTP_200_OK)
+        return Response(
+            dict(message="User does not exist"),
```

Comment:
would it be confusing for a user who knows they're in the system but have wrong credentials? probably have a message indicating the login wasn't successful.

Reply:
Working on it. Thanks.